ipfs-9-to-10/migration/config_conv.go

```diff
@@ -0,0 +1,205 @@
+package mg9
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/ipfs/fs-repo-migrations/stump"
+)
+
+var (
+	ip4BootstrapAddr  = "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+	quicBootstrapAddr = "/ip4/104.131.131.82/udp/4001/quic/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+)
+
+// convArray does an inplace conversion of an array of strings
+// configuration from one version to another
+type convArray func([]string) []string
+
+// convertFile converts a config file from one version to another
+func convertFile(orig string, new string, enableQuic bool, convBootstrap convArray, convSwarm convArray) error {
+	in, err := os.Open(orig)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(new)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	// Make sure file has 600 permissions
+	if err := out.Chmod(0600); err != nil {
```

Comment:
Ah, now I see what you mean. If we're _renaming_ the config, we don't need to chmod it (we'll just inherit the original mode).